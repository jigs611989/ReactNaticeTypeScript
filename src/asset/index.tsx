export { default as Color } from './Color';
export { default as AppStyle } from './AppStyle';
export { default as String } from './String';
export { default as Images } from './image';
