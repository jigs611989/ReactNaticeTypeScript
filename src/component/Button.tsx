import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import styles from './style/Button';

type Props = {
  title: string,
  onPress: () => void,
  style?: object,
  textStyle?: object,
}

const Button: React.FC<Props> = ({title, onPress, style, textStyle}) => {
  return (
    <TouchableOpacity onPress={onPress} style={[styles.button, style]}>
      <Text style={[styles.text, textStyle]}>{title}</Text>
    </TouchableOpacity>
  )
}

export default Button;
