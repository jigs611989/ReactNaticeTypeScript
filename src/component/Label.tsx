import React from 'react';
import { Text } from 'react-native';
import styles from './style/Label';

type Props = {
  title: string;
  style?: object;
};

const Label: React.FC<Props> = ({ style, title, ...props }) => (
  <Text 
    style={[styles.title, style]}
    {...props}
  >
    {title}
  </Text>
);

export default Label;
