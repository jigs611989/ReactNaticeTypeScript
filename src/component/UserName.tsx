import React from 'react';
import { View, Text } from 'react-native';
import { useSelector } from 'react-redux';
import style from './style/UserName';

const UserName = () => {
  const name = useSelector((state) => state.name);

  return (
    <View style={style.container}>
      <Text style={style.text}>{name}</Text>
    </View>
  );
};

export default UserName;
