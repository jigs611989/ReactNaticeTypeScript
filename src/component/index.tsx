export { default as Button } from './Button';
export { default as Label } from './Label';
export { default as UserName } from './UserName';
export { default as TextField } from './TextField';
export { default as SliderButton } from './SliderButton';
