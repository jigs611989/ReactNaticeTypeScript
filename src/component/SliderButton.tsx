import React from 'react';
import { Image, Text } from 'react-native';
import Slider from 'react-native-slide-to-unlock';
import { String } from '../asset';
import style from './style/SliderButton';

type Props = {
  onPress: () => void,
  title?: string,
  childrenContainer?: object,
  containerStyle?: object,
  sliderElement?: React.Component,
  testID?: string
};

const SliderButton: React.FC<Props> = ({
  onPress,
  title,
  sliderElement,
  childrenContainer,
  containerStyle,
  testID
}) => {
  return (
    <Slider
      childrenContainer={childrenContainer}
      onEndReached={onPress}
      containerStyle={[style.containerStyle, containerStyle]}
      sliderElement={
        sliderElement ? (
          sliderElement
        ) : (
          <Image
            style={style.imageStyle}
            source={{
              uri:
                'https://facebook.github.io/react-native/docs/assets/favicon.png',
            }}
          />
        )
      }>
      <Text>{title ? title : String.Buttons.slideMe}</Text>
    </Slider>
  );
};

export default SliderButton;
