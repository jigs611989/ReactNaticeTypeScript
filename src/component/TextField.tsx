import React from 'react';
import { TextInput, TextInputProps, View } from 'react-native';
import { Color } from '../asset';
import styles from './style/TextField';

type TextFieldProps = TextInputProps & {
  containerStyle?: object;
  inputStyle?: object;
};

const TextField: React.FC<TextFieldProps> = ({
  containerStyle,
  inputStyle,
  ...props
}) => {
  return (
    <View style={[styles.container, containerStyle]}>
      <TextInput
        placeholderTextColor={Color.placeholder}
        style={[styles.input, inputStyle]}
        {...props}
      />
    </View>
  );
};

export default TextField;
