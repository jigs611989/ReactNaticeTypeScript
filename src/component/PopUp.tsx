import React, {
  forwardRef,
  useCallback,
  useImperativeHandle,
  useState,
} from 'react';
import { Modal, View } from 'react-native';
import { BlurView } from '@react-native-community/blur';
import Label from './Label';
import Button from './Button';
import styles from './style/PopUp';

const Backdrop = () => (
  <BlurView style={styles.backdrop} blurType="light" blurAmount={5} />
);

type Props = {
  title: string;
  description?: string;
  onPress: () => void;
};

const PopUp = (props: Props, ref: any) => {
  const [isVisible, setIsVisible] = useState(false);

  const show = useCallback(() => setIsVisible(true), []);
  const hide = useCallback(() => setIsVisible(false), []);

  useImperativeHandle(ref, () => ({
    show,
    hide,
  }));

  const { title, description, onPress } = props;

  return (
    <Modal
      testID="ErrorPopupModal"
      visible={isVisible}
      onRequestClose={() => setIsVisible(false)}
      transparent={true}>
      <View style={styles.container}>
        <Backdrop />
        <View style={styles.popUp}>
          <Label title={title} style={styles.title} />
          {description ? (
            <Label title={description} style={styles.description} />
          ) : null}
          <Button title={'Okay'} onPress={onPress} style={styles.buttonStyle} />
        </View>
      </View>
    </Modal>
  );
};

export default forwardRef(PopUp);
