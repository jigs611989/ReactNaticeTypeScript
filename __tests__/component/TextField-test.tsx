import React from 'react';
import { TextField } from '../../src/component';
import { render } from '@testing-library/react-native';
import { fireEvent } from '@testing-library/react-native';

describe('TextField Component', () => {
  const mockonChangeText = jest.fn();
  const placeholder = 'Test placeholder prop';

  it('should have a placeholder', () => {
    const { getByPlaceholderText } = render(
      <TextField placeholder={placeholder} />,
    );
    const testInputObj = getByPlaceholderText(placeholder);
    expect(testInputObj).toBeDefined();
  });

  it('should call onChangeText when user is typing', () => {
    const { getByPlaceholderText } = render(
      <TextField placeholder={placeholder} onChangeText={mockonChangeText} />,
    );
    const testInputObj = getByPlaceholderText(placeholder);
    fireEvent(testInputObj, 'onChangeText', 'testing');
    expect(mockonChangeText).toHaveBeenCalledWith('testing');
  });

  it('should match the snapshot', () => {
    const { toJSON } = render(<TextField placeholder={placeholder} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
