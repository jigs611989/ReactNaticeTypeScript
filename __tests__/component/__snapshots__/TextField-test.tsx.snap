// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Component should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "rgba(11, 18, 30, 0.1)",
        "borderRadius": 10,
        "borderWidth": 1,
        "flexDirection": "row",
        "margin": 20,
        "padding": 16,
        "width": "80%",
      },
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    placeholder="Test placeholder prop"
    placeholderTextColor="#747577"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "color": "#101313",
          "flex": 1,
          "fontSize": 14,
          "lineHeight": 16,
          "padding": 0,
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
